lib/rules/valid-jsdoc.js

```diff
@@ -333,6 +307,42 @@ module.exports = {
                     }
                 });
 
+                if (hasParams) {
```

Comment:
I think the `hasParams` variable is unnecessary. If there are no `param` tags, then the `hasParamsTags` array will be empty, so the `forEach` loop will just do nothing anyway.